File: blog/src/main/java/com/github/blog/controller/util/marker/BaseMarker.java
package com.github.blog.controller.util.marker;

import jakarta.validation.GroupSequence;
import jakarta.validation.groups.Default;

/**
 * @author Raman Haurylau
 */
public interface BaseMarker {
    interface First {
    }

    interface Second {
    }

    interface Third {
    }

    interface Create {
    }

    interface Update {
    }

    @GroupSequence({Create.class, Default.class})
    interface onCreate {
    }

    @GroupSequence({Update.class, Default.class})
    interface onUpdate {
    }
}

File: api-gateway/src/main/kotlin/com/github/api_gateway/controller/util/marker/BaseMarker.kt
package com.github.api_gateway.controller.util.marker

import jakarta.validation.GroupSequence
import jakarta.validation.groups.Default

/**
 * @author Raman Haurylau
 */
interface BaseMarker {
    interface First
    interface Second
    interface Third

    interface Create
    interface Update

    @GroupSequence(value = [Create::class, Default::class])
    interface onCreate

    @GroupSequence(value = [Update::class, Default::class])
    interface onUpdate
}